Docs/(sakuli/sakuli dashboard#60)/how to deploy dashboard (#61)
sakuli/docs: content/enterprise_features/sakuli-dashboard-openshift.md

---
title : "Sakuli Dashboard on Openshift"
date :  2019-09-12T14:16:46+02:00
weight : 8
---

# Sakuli Dashboard on Openshift

> Once you obtained a Sakuli Enterprise license your docker-user will be granted access to the private Sakuli dashboard image.

Reviewer: ```suggestion
**Attention:** Only Sakuli Enterprise users will be eligible to access the private Sakuli Dashboard Docker images.
```

## Prerequisites

> To setup a dashboard container on your OpenShift cluster,
> you have to import the images from `taconsol/sakuli-dashboard`.
> In order to authenticate to docker.io during build,
> you have to create a docker registry secret with your
> `<docker-username>`
> and `<docker-password>`.

Reviewer: ```suggestion
A Sakuli Dashboard setup on your OpenShift cluster requires you to import the Sakuli Dashboard image from Docker Hub. In order to do so, please configure a secret storing your `<docker-username>` and `<docker-password>` and add it to your service account for authentication.
```

{{<highlight bash>}}
oc create secret docker-registry dockerhub-sakuli-secret \
 --docker-server=docker.io \
 --docker-username=<docker-username> \
 --docker-password=<docker-password> \
 --docker-email=unused
{{</highlight>}}

Enable access to the secret from the default service account:

{{<highlight bash>}}
oc secrets link default dockerhub-sakuli-secret --for=pull
{{</highlight>}}

Now you can import the image:

{{<highlight bash>}}
oc import-image sakuli-dashboard \
 --from=docker.io/taconsol/sakuli-dashboard \
 --confirm \
 --scheduled=true \
 --all=true
{{</highlight>}}

When using OpenShift v3.10 or v3.11, add the `--reference-policy=local` flag:

{{<highlight bash>}}
oc import-image sakuli-dashboard \
 --from=docker.io/taconsol/sakuli-dashboard \
 --confirm \
 --scheduled=true \
 --all \
 --reference-policy=local
{{</highlight>}}

*Note: The `oc import-image` statement is configured to not only import all available sakuli-dashboard images but also to
check for updates automatically.*

> You can start your Sakuli dashboard using two different approaches:
> - Start the Sakuli dashboard via CLI with [oc](#oc).
> - Start the Sakuli dashboard using a ready to use [template](#dashboard-template).

Reviewer: Aren't we also using `oc` to apply the template? 😄 

```suggestion
You can start your Sakuli Dashboard using two different approaches:
- Set up the Sakuli Dashboard manually via [CLI](#oc-cli).
- Set up the Sakuli Dashboard using a ready to use [template](#oc-dashboard-template).
```

> ## Starting the Sakuli dashboard with oc {#oc}

Reviewer: ```suggestion
## Set up the Sakuli Dashboard manually via CLI {#oc-cli}
```

> To create the application with the openshift client, you can use the `new-app` command. You can add environment variables
> via the `-e` parameter.

Reviewer: ```suggestion
An easy way to set up a Sakuli Dashboard is by using `oc new-app`. Required environment variables can be passed along via the `-e` parameter.
```

{{<highlight bash>}}
oc new-app sakuli-dashboard \ 
 -e DASHBOARD_CONFIG="${DASHBOARD_CONFIG}"  \
 -e ACTION_CONFIG="${ACTION_CONFIG}"  \
 -e CLUSTER_CONFIG="${CLUSTER_CONFIG}"  \
 -e CRONJOB_CONFIG="${CRONJOB_CONFIG}" \
 -e SAKULI_LICENSE_KEY="${SAKULI_LICENSE_KEY}"
{{</highlight>}}

> Now you need to expose your service, to make it available outside your cluster.

Reviewer: ```suggestion
Once completed, the service has to be exposed to make available from outside the cluster.
```

{{<highlight bash>}}
oc expose svc/sakuli-dashboard
{{</highlight>}}

> ## Starting the Sakuli dashboard with a template {#dashboard-template}

Reviewer: ```suggestion
## Set up the Sakuli Dashboard using a ready to use template {#oc-dashboard-template}

> An XL Sakuli license key is required to start the Sakuli dashboard. You can add your license key with a secret, which is
> then referenced to the deployment in the template.

Reviewer: ```suggestion
The Sakuli Dashboard OpenShift template references the Sakuli license key via a secret.
So before applying the template, please add your license key secret first.
```

{{<highlight bash>}}
oc create secret generic sakuli-license-key \
 --from-literal="SAKULI_LICENSE_KEY=${SAKULI_LICENSE_KEY}"
{{</highlight>}}

> The following config template is ready to use to configure and deploy your Sakuli dashboard.

Reviewer: ```suggestion
The following ready-to-use template will configure a Sakuli Dashboard and deploy it to your cluster
```

> Just copy and save the dashboard template bellow as `dashboard-template.yml`. Furthermore, add your dashboard, action, 

Reviewer: ```suggestion
Just copy and save the dashboard template below as `dashboard-template.yml`. Furthermore, add your dashboard, action, 
```

> cluster and cronjob configurations listed under ConfigMap as well as the image tag.

Reviewer: There are additional things to provide to the template:

```
#change the address to your openshift registry, your namespace and the image tag you want to use
image: ${OPENSHIFT_REGISTRY_ADDRESS}/${NAMESPACE}/sakuli-dashboard:${IMAGE_TAG}
```
